common/lib/dependabot/git_metadata_fetcher.rb

```diff
@@ -48,7 +48,7 @@ def head_commit_for_ref(ref)
         # Remove the opening clause of the upload pack as this isn't always
         # followed by a line break. When it isn't (e.g., with Bitbucket) it
         # causes problems for our `sha_for_update_pack_line` logic
-        line = upload_pack.gsub(/.*git-upload-pack/, "").
+        line = upload_pack.gsub(/^[0-9a-f]{4}# service=git-upload-pack/, "").
```

Comment:
Could you add a comment to explain this new prefix (maybe with a link to the docs)? If I was to read this Regex without the context of this pull request, I'd be a little lost re: why it's there.

Reply:
Thanks @landongrindheim, you're right. A added a comment about the format, what do you think?

Reply:
🎯 It's perfect!